clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3034,28 +3042,9 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
   }
 
   bool handleSyclSpecialType(ParmVarDecl *PD, QualType ParamTy) final {
-    if (SemaSYCL::isSyclType(ParamTy, SYCLTypeAttr::work_group_memory) ||
-        SemaSYCL::isSyclType(ParamTy,
-                             SYCLTypeAttr::dynamic_work_group_memory)) {
-      const auto *RecordDecl = ParamTy->getAsCXXRecordDecl();
-      assert(RecordDecl && "The type must be a RecordDecl");
-      CXXMethodDecl *InitMethod = getMethodByName(RecordDecl, InitMethodName);
-      assert(InitMethod && "The type must have the __init method");
-      // Don't do -1 here because we count on this to be the first parameter
-      // added (if any).
-      size_t ParamIndex = Params.size();
-      for (const ParmVarDecl *Param : InitMethod->parameters()) {
-        QualType ParamTy = Param->getType();
-        addParam(Param, ParamTy.getCanonicalType());
-        // Propagate add_ir_attributes_kernel_parameter attribute.
-        if (const auto *AddIRAttr =
-                Param->getAttr<SYCLAddIRAttributesKernelParameterAttr>())
-          Params.back()->addAttr(AddIRAttr->clone(SemaSYCLRef.getASTContext()));
-      }
-      LastParamIndex = ParamIndex;
-    } else // TODO
-      unsupportedFreeFunctionParamType();
-    return true;
+    if (SemaSYCL::isSyclType(ParamTy, SYCLTypeAttr::accessor))
+      unsupportedFreeFunctionParamType(); // TODO
```

Comment:
I don't really like that we keep using `unsupportedFreeFunctionParamType` because it will say
`backend error: Only scalars and pointers are permitted as free function parameters ` once triggered, which is not true for a long time. Can we simply use `lllvm_unreacheable("some message)"` here for now, since `SyclKernelFieldChecker` is supposed to guarantee to us that we don't enter any visitors after it once it failed?